test/e2e/autoscaler.go

```diff
@@ -239,67 +298,127 @@ func AutoscalerSpec(ctx context.Context, inputGetter func() AutoscalerSpecInput)
+			})
+
+			By("Checking the MachinePool is scaled down")
+			// Since we scaled up the MachinePool manually and the workload has not changed auto scaler
+			// should detect that there are unneeded nodes and scale down the MachinePool.
+			framework.AssertMachinePoolReplicas(ctx, framework.AssertMachinePoolReplicasInput{
+				Getter:             input.BootstrapClusterProxy.GetClient(),
+				MachinePool:        clusterResources.MachinePools[0],
+				Replicas:           mpScaledUpReplicas,
+				WaitForMachinePool: input.E2EConfig.GetIntervals(specName, "wait-controllers"),
+			})
+
+			if input.ScaleToAndFromZero {
+				By("Enabling autoscaler for the MachinePool to zero")
+				// Enable autoscaler on the MachinePool.
+				framework.EnableAutoscalerForMachinePoolTopologyAndWait(ctx, framework.EnableAutoscalerForMachinePoolTopologyAndWaitInput{
+					ClusterProxy:                input.BootstrapClusterProxy,
+					Cluster:                     clusterResources.Cluster,
+					NodeGroupMinSize:            "0",
+					NodeGroupMaxSize:            mpNodeGroupMaxSize,
+					WaitForAnnotationsToBeAdded: input.E2EConfig.GetIntervals(specName, "wait-autoscaler"),
+				})
+
+				// We can savely assume that mdReplicas is 1.
+				var mdReplicas int32 = 1
+
+				By("Scaling the MachinePool scale up deployment to 1")
+				framework.ScaleScaleUpDeploymentAndWait(ctx, framework.ScaleScaleUpDeploymentAndWaitInput{
+					ClusterProxy: workloadClusterProxy,
+					// Set replicas to 1, because we still have 1 Machine from MachineDeployments.
+					Replicas: 1,
+				}, input.E2EConfig.GetIntervals(specName, "wait-autoscaler")...)
+
+				By("Checking the MachinePool finished scaling down to zero")
+				framework.AssertMachinePoolReplicas(ctx, framework.AssertMachinePoolReplicasInput{
+					Getter:             input.BootstrapClusterProxy.GetClient(),
+					MachinePool:        clusterResources.MachinePools[0],
+					Replicas:           mdReplicas,
+					WaitForMachinePool: input.E2EConfig.GetIntervals(specName, "wait-controllers"),
+				})
+
+				By("Scaling the MachinePool scale up deployment to 2")
+				framework.ScaleScaleUpDeploymentAndWait(ctx, framework.ScaleScaleUpDeploymentAndWaitInput{
+					ClusterProxy: workloadClusterProxy,
+					// Set replicas to 2, because we still have 1 Machine from MachineDeployments.
+					Replicas: mdReplicas + 1,
+				}, input.E2EConfig.GetIntervals(specName, "wait-autoscaler")...)
+
+				By("Checking the MachineDeployment finished scaling up")
+				framework.AssertMachinePoolReplicas(ctx, framework.AssertMachinePoolReplicasInput{
+					Getter:             input.BootstrapClusterProxy.GetClient(),
+					MachinePool:        clusterResources.MachinePools[0],
+					Replicas:           1,
+					WaitForMachinePool: input.E2EConfig.GetIntervals(specName, "wait-controllers"),
+				})
+			}
+
+			By("Disabling the autoscaler for MachineDeployments to test MachinePools")
```

Comment:
Wondering why we do this at this point

Reply:
missed fixing this string :-)

These last two steps are not really necessary 👍  I'll remove them